drltrace/drltrace.cpp

```diff
@@ -79,6 +80,276 @@ static file_t outf;
+    tokens_count = split(line, ",", tokens);
+
+    if (tokens_count <= 0) {
+        VNOTIFY(1, "unable to parse config file at line %d: %s" NL, line_num, line);
+        delete (tokens);
+        return false;
+    }
+
+    std::vector<drsys_arg_t *> *args_vector = new std::vector<drsys_arg_t *>();
+    std::vector<char*>::iterator it;
+    for (it = tokens->begin(); it != tokens->end(); it++) {
+        /* i#1948: Currently, we don't support ret value printing and
+         * skipping it here. */
+        if (elem_index >= 2) {
+            tmp_arg = config_parse_type(*it, elem_index - 2);
+            args_vector->push_back(tmp_arg);
+        } else if (elem_index == 1)
+            func_name = *it;
+
+        elem_index++;
+    }
+
+    delete tokens;
+
+    if (func_name == NULL || args_vector->size() <= 0) {
+        VNOTIFY(1, "unable to parse config file at line %d: %s" NL, line_num, line);
+        return false;
+    }
+
+    VNOTIFY(2, "adding %s from config file with %d arguments in the hashtable" NL,
+            func_name, args_vector->size());
+    IF_DEBUG(bool ok =)
+        libcalls_hashtable_insert(strdup(func_name), args_vector);
+    ASSERT(ok, "failed to add libcall in the hashtable");
+
+    return true;
+}
+
+static void
+parse_config(void) {
+    uint64 size_to_read = 0;
+    size_t size_read = 0;
+    file_t file_desc = INVALID_FILE;
+    int lines_count = 0, line_num = 1;
+    char *buf = NULL;
+
+    if (op_disable_config.get_value() == true)
+        return;
+
+    /* open and read config file */
+    file_desc = dr_open_file(op_config_file.get_value().c_str(), DR_FILE_READ);
+    if (file_desc == INVALID_FILE) {
+        VNOTIFY(0, "unable to open config file at %s, config is not used" NL,
+                    op_config_file.get_value().c_str());
+        op_disable_config.set_value(true);
+        return;
+    }
+
+    if (dr_file_size(file_desc, &size_to_read) == false) {
+        VNOTIFY(0, "unable to open config file at %s, config is not used" NL,
+                    op_config_file.get_value().c_str());
+        op_disable_config.set_value(true);
+        dr_close_file(file_desc);
+        return;
+    }
+
+    buf = (char *)global_alloc(size_to_read, HEAPSTAT_MISC);
+
+    size_read = dr_read_file(file_desc, buf, size_to_read);
+    dr_close_file(file_desc);
+    if (size_read <= 0) {
+        VNOTIFY(0, "unable to open config file at %s, config is not used" NL,
+                    op_config_file.get_value().c_str());
+        global_free(buf, size_to_read, HEAPSTAT_MISC);
+        op_disable_config.set_value(true);
+        return;
+    }
+    buf[size_read - 1] = '\0';
+
+    std::vector<char *> *lines_list = new std::vector<char *> ();
+    lines_count = split(buf, "\r\n", lines_list); /* split buffer by lines */
+
+    global_free(buf, size_to_read, HEAPSTAT_MISC);
+
+    if (lines_count <= 0) {
+        VNOTIFY(1, "An empty config file was specified, config is not used" NL);
```

Comment:
No, it turns that all these messages are not visible in release build. I will fix it in a separate CL, added comment for now.